File: Reflow/Helper/IColumnTypeTranslator.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.OleDb;
using System.Linq;
using System.Text;

namespace Reflow.Helper
{
    public interface IColumnTypeTranslator
    {
        DbType GetDbTypeFromSystemTypeString(string systemType);
        
        string GetDataTypeExpression(ColumnDefinition dataType);

        string GetAnsiStringExpression(ColumnDefinition dataType);
        string GetAnsiStringFixedLengthExpression(ColumnDefinition dataType);
        string GetBinaryExpression(ColumnDefinition dataType);
        string GetBooleanExpression(ColumnDefinition dataType);
        string GetByteExpression(ColumnDefinition dataType);
        string GetCurrencyExpression(ColumnDefinition dataType);
        string GetDateTimeExpression(ColumnDefinition dataType);
        string GetDecimalExpression(ColumnDefinition dataType);
        string GetDoubleExpression(ColumnDefinition dataType);
        string GetInt16Expression(ColumnDefinition dataType);
        string GetInt32Expression(ColumnDefinition dataType);
        string GetInt64Expression(ColumnDefinition dataType);
        string GetSingleExpression(ColumnDefinition dataType);
        string GetStringExpression(ColumnDefinition dataType);
        string GetStringFixedLengthExpression(ColumnDefinition dataType);
        string GetUInt16Expression(ColumnDefinition dataType);
        string GetUInt32Expression(ColumnDefinition dataType);
        string GetUInt64Expression(ColumnDefinition dataType);
        string GetGuidExpression(ColumnDefinition dataType);

        char OpeningIdentifier { get; }
        char ClosingIdentifier { get; }
        char ParameterIdentifier { get; }
        char ParameterPlaceholderIdentifier { get; }

        string NextStatementIdentifier { get; }
        string IdentityExpression { get; }
        bool SupportsIdentifierAroundTypename { get; }
        string GetDefaultExpression(string defaultValue, bool needsQuote);

    }
}

File: Java/src/main/java/com/nuarca/etl/helper/IColumnTypeTranslator.java
//
// Translated by CS2J (http://www.cs2j.com): 12/6/2017 2:34:20 PM
//

package com.nuarca.etl.helper;

import com.nuarca.etl.ColumnDefinition;

import java.sql.SQLType;

public interface IColumnTypeTranslator   
{
    SQLType getDbTypeFromSystemTypeString(String systemType) throws Exception ;

    String getDataTypeExpression(ColumnDefinition dataType) throws Exception ;

    String getAnsiStringExpression(ColumnDefinition dataType) throws Exception ;

    String getAnsiStringFixedLengthExpression(ColumnDefinition dataType) throws Exception ;

    String getBinaryExpression(ColumnDefinition dataType) throws Exception ;

    String getBooleanExpression(ColumnDefinition dataType) throws Exception ;

    String getByteExpression(ColumnDefinition dataType) throws Exception ;

    String getCurrencyExpression(ColumnDefinition dataType) throws Exception ;

    String getDateTimeExpression(ColumnDefinition dataType) throws Exception ;

    String getDecimalExpression(ColumnDefinition dataType) throws Exception ;

    String getDoubleExpression(ColumnDefinition dataType) throws Exception ;

    String getInt16Expression(ColumnDefinition dataType) throws Exception ;

    String getInt32Expression(ColumnDefinition dataType) throws Exception ;

    String getInt64Expression(ColumnDefinition dataType) throws Exception ;

    String getSingleExpression(ColumnDefinition dataType) throws Exception ;

    String getStringExpression(ColumnDefinition dataType) throws Exception ;

    String getStringFixedLengthExpression(ColumnDefinition dataType) throws Exception ;

    String getUInt16Expression(ColumnDefinition dataType) throws Exception ;

    String getUInt32Expression(ColumnDefinition dataType) throws Exception ;

    String getUInt64Expression(ColumnDefinition dataType) throws Exception ;

    String getGuidExpression(ColumnDefinition dataType) throws Exception ;

    char getOpeningIdentifier() throws Exception ;

    char getClosingIdentifier() throws Exception ;

    char getParameterIdentifier() throws Exception ;

    char getParameterPlaceholderIdentifier() throws Exception ;

    String getNextStatementIdentifier() throws Exception ;

    String getIdentityExpression() throws Exception ;

    boolean getSupportsIdentifierAroundTypename() throws Exception ;

    String getDefaultExpression(String defaultValue, boolean needsQuote) throws Exception ;

}


